### PR TITLE
OpenOxygenMeter: Add negative flow calibration routine

### PR DESCRIPTION
**Pull request by frauzufall**
This adds the negative flow calibration to 100% O2 routine, as discussed in https://github.com/beniroquai/Open-Oxygen-Flowsensor/pull/10. It needs to be properly tested.

### CODE REVIEW

File: ARDUINO/OpenOxygenFlowMeter/OpenOxygenFlowMeter.ino [hunk 1]
```diff
@@ -15,14 +16,47 @@ int16_t adc0 = -1;
...
+
+unsigned static long infoMsgDelay = 1000 * 2; // 2s delay to display an info message
+
+uint8_t countdown = 0;
+
+float flowRate = 0;
+float diffPressure = 0;
+float oxygenLevel = 0;
+
+enum DeviceState {
```
Comment (beniroquai) [inline ARDUINO/OpenOxygenFlowMeter/OpenOxygenFlowMeter.ino:45]: Soo fancy!! :)
I like that!

File: ARDUINO/OpenOxygenFlowMeter/OpenOxygenFlowMeter.ino [hunk 2]
```diff
@@ -15,14 +16,47 @@ int16_t adc0 = -1;
...
+float adcToMV =  0.1875;
 
+// variables for initial auto calibration to air
+float oxyCalibration = 20.9;
 boolean shouldDoWarmupCalibration = true; //change this to false to skip warmup / calibration phase
 unsigned long startTime;
 unsigned static long warmupDelayInMs = 1000 * 10; // 10s warmup time before calibrating to 21% oxygen level
 
+// variables for negative flow calibration
+float minAbsDiffPressure = 2; // minimal absolute value of pressure to trigger negative flow calibration
```
Comment (beniroquai) [inline ARDUINO/OpenOxygenFlowMeter/OpenOxygenFlowMeter.ino:28]: This we have to tune, right? Perhaps we could set this a bit higher to make sure it's not some "wind"/noise?

**Review by beniroquai — APPROVED** [review]
Really nice addition to the code! I'll check that on the device later today! :)

**Comment by beniroquai** [comment]
I will test it today. Though I'm not sure if I can get `100%` oxygen. But it should be not too difficult to change the code to alternatively calibrate it with `0%` O2 level.
I'll get back to you! Thanks a lot for your work! :-)

File: ARDUINO/OpenOxygenFlowMeter/OpenOxygenFlowMeter.ino (hunk 2 above)
Comment (frauzufall) [reply ARDUINO/OpenOxygenFlowMeter/OpenOxygenFlowMeter.ino:28]: Yes! I set it to 10 [here](https://github.com/beniroquai/Open-Oxygen-Flowsensor/pull/13) for now - please adjust it to something reasonable if you test it!

**Comment by frauzufall** [comment]
With the next PR I made it easier to adjust the target concentration of the negative flow routine, but we have to talk about calibrating to 0% - currently calibration (both using air after launch and using negative flow) is not adjusting `oxy_b`, only `oxy_m`. Therefore calibration to 0% will not work, right? In this case, we would have to keep the calibration of air at the launch and use both points, no? Let's talk about it later :+1:

**Pull request merged**